### PR TITLE
Declare variable t_start discrete in Modelica.StateGraph.Interfaces.PartialTransition

### PR DESCRIPTION
**Pull request by beutlich**
Avoid warning in Wolfram SystemModeler.

### CODE REVIEW

**Comment by henrikt-ma** [comment]
Why not initialize to `time` rather than `0`?

**Comment by beutlich** [comment]
> 
> 
> Why not initialize to `time` rather than `0`?

I tossed a coin, since not relevant anyway. But will do.

**Comment by henrikt-ma** [comment]
I note that this model seems to have the interesting problem of sometimes dealing with a discrete variable that doesn't change.  I suppose it would also be possible to express this more clearly as:
```
initial equation
  pre(t_start) = time;
…
if enableTimer then
  …
else
  when false then
    t_start = time; /* Dummy equation for t_start. */
  end when;
```
Or is there some detail about initializing {{pre(t_start)}} vs {{t_start}} that I'm missing?

**Comment by beutlich** [comment]
LGTM. I applied your proposed changes.

**Comment by henrikt-ma** [comment]
It just occurred to me that the canonical way of providing a dummy equation for a `discrete` variable should probably be this:
```
    when false then
      t_start = pre(t_start); /* Dummy equation for t_start */
    end when;
```
If this would be recognized as just that, the comment wouldn't even be necessary.

File: Modelica/StateGraph.mo [hunk 1]
```diff
@@ -2040,7 +2038,9 @@ package Interfaces "Connectors and partial models"
       t = if enableFire then t_dummy else 0;
       fire = enableFire and time >= t_start + waitTime;
     else
-      t_start = 0;
+      when false then
+        t_start = time; /* Dummy equation for t_start */
```
Comment (beutlich) [inline Modelica/StateGraph.mo]:
```suggestion
        t_start = pre(t_start);
```

**Comment by beutlich** [comment]
@HansOlsson OK for you to merge?

**Pull request merged**